### ISSUE TITLE
Doesn't render children in React 16.0

### ISSUE THREAD

**Issue by swarnakumar** [jasonslyvia/react-lazyload#125]
I have a component:

`<Lazyload once height={200}><Child/></Lazyload>`

This works in React 15.6.1, but not in React 16.0. The Child component's ComponentWillMount lifecycle method isn't being called. Neither is the render method.

I tried making once=false, still doesn't work.

**Comment by jasonslyvia** [comment]
I will look into this.

**Comment by kochis** [comment]
Can confirm, also seeing an issue on react 16. Looks like it's not being server-side rendered (just does shallow render). It does seem to be working fine on the client though (after mount)

**Comment by swarnakumar** [comment]
I am unable to make it work on client side either.

@kochis can you paste some code so I can figure out how you made it work?

**Comment by WendellLiu** [comment]
It doesn't work in my project on client side too.
And it work in version 15.6.1.

**Comment by jasonslyvia** [comment]
Hi guys, I just came back from a vacation and React 16 refactor is on the way, hopefully it won't be too late.

**Comment by WendellLiu** [comment]
thank you @jasonslyvia

**Comment by ryanmcafee** [comment]
Any update on this issue?

**Comment by jasonslyvia** [comment]
Hopefully I will push some commits around this weekend to solve this issue, stay tuned @ryanmcafee

**Comment by aFcFzF** [comment]
@jasonslyvia  thank you! ^_^

**Comment by fluse** [comment]
@jasonslyvia can you add more maintainer, to solve problems faster?

**Comment by jasonslyvia** [comment]
Sure, I'm always looking forward to maintainers, you in?

**Comment by modosc** [comment]
@jasonslyvia also happy to help out

**Comment by jasonslyvia** [comment]
Hi guys, I try to update React@16 and, ummm, everything seems working fine?  https://github.com/jasonslyvia/react-lazyload/tree/feat/react-16

**Comment by elrumordelaluz** [comment]
I'm on `react@16.0.0` and everything works as expected regarding `react-lazyload`. 
Only the usual warnings:

```
warning "react-lazyload@2.2.7" has incorrect peer dependency "react@^0.14.0 || ^15.0.0".
warning "react-lazyload@2.2.7" has incorrect peer dependency "react-dom@^0.14.0 || ^15.0.0".
```

**Comment by jasonslyvia** [comment]
thanks for confirming on this @elrumordelaluz

**Comment by elrumordelaluz** [comment]
want a PR to update `peer-deps` including `^16.0.0`?

**Comment by jasonslyvia** [comment]
there is already a pending one #126, I just want to make sure everything is fine since so many people  'reporting' this issue.

**Comment by jasonslyvia** [comment]
If everybody is okay, then I'm going to merge #126 and publish a new minor version.

**Comment by jasonslyvia** [comment]
I've published `2.3.0` with `react: "^16.0.0"` and `react-dom: "^16.0.0"` as peer dependencies

**Comment by swarnakumar** [comment]
Apologies for the late reply. The new version works. Thanks a lot.

**Comment by mrasoahaingo** [comment]
Any date for the release?

**Comment by elrumordelaluz** [comment]
Seems solved in `@2.3.0`
